### PR TITLE
Agregate formula data into algolia for web search

### PR DESCRIPTION
**Pull request by SMillerDev**
I have no idea what I'm doing, this doesn't seem like the best way to pass the API key and app ID but I couldn't figure out a better way.

### CODE REVIEW

File: script/generate.rb [hunk 1]
```diff
@@ -19,3 +32,10 @@
   IO.write("_data/cask/#{json_filename}", JSON.pretty_generate(c.to_h))
   FileUtils.cp "_api_formula.json.in", "api/cask/#{json_filename}"
 end
+
+require 'algoliasearch'
+Algolia.init(application_id: ARGV[0],
```
Comment (MikeMcQuaid) [inline script/generate.rb]: I reckon we can hardcode the `application_id` here.

File: script/generate.rb [hunk 2]
```diff
@@ -19,3 +32,10 @@
...
+             api_key:        ARGV[1])
+index = Algolia::Index.new('formula')
+index.clear_index()
+index.add_objects(formulae_indexes)
```
Comment (MikeMcQuaid) [inline script/generate.rb]: Add a newline.

File: script/generate.rb [hunk 3]
```diff
@@ -19,3 +32,10 @@
   IO.write("_data/cask/#{json_filename}", JSON.pretty_generate(c.to_h))
   FileUtils.cp "_api_formula.json.in", "api/cask/#{json_filename}"
 end
+
+require 'algoliasearch'
+Algolia.init(application_id: ARGV[0],
+             api_key:        ARGV[1])
```
Comment (MikeMcQuaid) [inline script/generate.rb]: Using `ENV["ALGOLIA_API_KEY"]` directly seems fine.

File: script/generate.rb [hunk 4]
```diff
@@ -11,6 +12,18 @@
   html = IO.read "_formula.html.in"
   html.gsub!("title: $TITLE", "title: #{f.name}")
   IO.write("formula/#{f.name}.html", html)
+  formulae_indexes << {
```
Comment (MikeMcQuaid) [inline script/generate.rb]: Can this use `f.to_hash` instead or does it need to be flat?

**Comment by SMillerDev** [comment]
@MikeMcQuaid pushed. Someone will also have to do https://www.algolia.com/doc/faq/accounts-billing/i-m-an-open-source-developer/ and it might be better to have someone in a leadership like role do that.

**Comment by MikeMcQuaid** [comment]
> @MikeMcQuaid pushed. Someone will also have to do https://www.algolia.com/doc/faq/accounts-billing/i-m-an-open-source-developer/ and it might be better to have someone in a leadership like role do that.

@SMillerDev Done! Will report back when I hear anything.

**Comment by MikeMcQuaid** [comment]
Closing in favour of https://github.com/Homebrew/formulae.brew.sh/pull/103 https://github.com/Homebrew/brew.sh/pull/326

**Pull request closed without merging**